versioned_docs/version-2.x/magic-button/contribution-mode.mdx

```diff
@@ -0,0 +1,130 @@
+---
+sidebar_position: 1
+title: Contribution Mode
+description:
+  Contribution Mode is the core of Magic Button, streamlining server-client
+  interaction for collaborative and efficient content creation.
+---
+
+import SinceVersion from "@site/src/components/SinceVersion";
+
+<SinceVersion tag="2.25" />
+
+<p>{frontMatter.description}</p>
+
+## How to implement the Magic Button Component
+
+To add the **Magic Button** to your application you simply need to update your
+`.front-commerce.js` configuration:
+
+```js title=".front-commerce.js"
+module.exports = {
+  name: "Front Commerce DEV",
+  url: "http://localhost:4000",
+  modules: [
+    "./theme-chocolatine",
+    // add-next-line
+    "./node_modules/front-commerce/modules/magic-button",
+  ],
+  serverModules: [
+    { name: "FrontCommerce", path: "server/modules/front-commerce" },
+    { name: "Magento2", path: "server/modules/magento2" },
+  ],
+  webModules: [
+    { name: "FrontCommerce", path: "./src/web" },
+    { name: "ThemeChocolatine", path: "./theme-chocolatine/web" },
+  ],
+};
+```
+
+The **Magic Button** is only visible when the **Contribution Mode** is active.
+
+## How to activate Contribution Mode
+
+### Activate with Config Provider
+
+The default activation for **Contribution Mode** is defined by a the
+[contributionModeConfigProvider](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/e79a04bf0fd4c171b1530e67f7ac1b53cabed83a/src/server/modules/Front-Commerce/Contribution-Mode/contribution-mode/contributionModeConfigProvider.js).
+
+You can follow the
+[Override an existing configuration](/docs/2.x/advanced/server/configurations#override-an-existing-configuration)
+to update the value.
+
+:::caution
+
+While in development you can use the
+`FRONT_COMMERCE_CONTRIBUTION_MODE_FORCE=true` environment variable to force the
+contribution mode to be enabled, but this should not be used in production.
+
+:::
+
+### Activate with Loader
+
+To activate the **Contribution Mode** through a loader, you can take advantage
+of the `forceEnablePreviewMode` option in the
+[ContributionModeLoader](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/7a2aee7e11c7bd45cfc3a5a940407f9e220e66f4/src/server/modules/Front-Commerce/Contribution-Mode/contribution-mode/loaders/ContributionModeLoader.js#L57-60)
+
+```js
+export default {
+    namespace:"MyModule"
+    dependencies:["Front-Commerce/Contribution-Mode"]
+    resolvers:{
+        Mutation:{
+            myQuery: async (root, args, context, info) => {
```

Comment:
**issue:** we usually don't mention `info` as it contains information for power users only.

**suggestion:** remove the last params from all examples on this page.

Also, the mutation name is weird _(query term used)_ and not very realistic:
```suggestion
            enableContributorMode: async (root, args, context) => {
              const isEditor = await context.loader.AcmeUser.isContentEditor();
               if (!isEditor) {
                 return false;
               }
```